File: Difficulty: Easy/Missing in Array/missing-in-array.py
class Solution:
    def missingNum(self, arr):
        # code here
        res = [-1] * (len(arr) + 2)
        
        for num in arr:
            res[num] = num
            
        for i in range(1,len(res)):
            if res[i] == -1:
                return i
                

File: Difficulty: Easy/Missing in Array/missing-in-array.cpp

class Solution {
  public:
    int missingNum(vector<int>& arr) {
        // code here
        int n = arr.size();
        vector<int> res(n,-1);
        
        sort(arr.begin(),arr.end());
        
        for(int i = 0;i < n;++i){
            if (arr[i] != i+1){
                return i+1;
            }
        }
        return arr[n-1] + 1;
        
        
    }
};